src/tool_operate.c

```diff
@@ -808,7 +851,13 @@ static CURLcode ipfs_url_rewrite(CURLU *uh, const char *protocol, char **url,
    * if we do have something but if it's an invalid url.
    */
   if(config->ipfs_gateway) {
-    if(curl_url_set(ipfsurl, CURLUPART_URL, config->ipfs_gateway,
+    /* ensure the gateway ends in a trailing / */
+    if(ensure_trailing(&config->ipfs_gateway, '/') != CURLE_OK) {
+      result = CURLE_OUT_OF_MEMORY;
+      goto clean;
+    }
+
+    if(curl_url_set(gatewaysurl, CURLUPART_URL, config->ipfs_gateway,
                     CURLU_GUESS_SCHEME)
                     == CURLUE_OK) {
```

Comment:
*I* would write these conditions like this to make them slightly shorter and in my view easier to read:

```suggestion
    if(!curl_url_set(gatewaysurl, CURLUPART_URL, config->ipfs_gateway,
                     CURLU_GUESS_SCHEME)) {
```